rpc/src/rpc_trait.rs

```diff
@@ -123,36 +122,36 @@ pub trait ProtectedRpcFunctions {
     //     nightly,
     //     doc(include = "../documentation/private_endpoints/createtransactionkernel.md")
     // )]
-    fn create_transaction_kernel(&self, transaction_input: TransactionInputs) -> Result<String, RpcError>;
+    async fn create_transaction_kernel(&self, transaction_input: TransactionInputs) -> Result<String, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(nightly, doc(include = "../documentation/private_endpoints/createtransaction.md"))]
-    fn create_transaction(&self, transaction_kernel: String) -> Result<CreateRawTransactionOuput, RpcError>;
+    async fn create_transaction(&self, transaction_kernel: String) -> Result<CreateRawTransactionOuput, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(nightly, doc(include = "../documentation/private_endpoints/getrecordcommitments.md"))]
-    fn get_record_commitments(&self) -> Result<Vec<String>, RpcError>;
+    async fn get_record_commitments(&self) -> Result<Vec<String>, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(
     //     nightly,
     //     doc(include = "../documentation/private_endpoints/getrecordcommitmentcount.md")
     // )]
-    fn get_record_commitment_count(&self) -> Result<usize, RpcError>;
+    async fn get_record_commitment_count(&self) -> Result<usize, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(nightly, doc(include = "../documentation/private_endpoints/getrawrecord.md"))]
-    fn get_raw_record(&self, record_commitment: String) -> Result<String, RpcError>;
+    async fn get_raw_record(&self, record_commitment: String) -> Result<String, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(nightly, doc(include = "../documentation/private_endpoints/decoderecord.md"))]
-    fn decode_record(&self, record_bytes: String) -> Result<RecordInfo, RpcError>;
+    async fn decode_record(&self, record_bytes: String) -> Result<RecordInfo, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(nightly, doc(include = "../documentation/private_endpoints/decryptrecord.md"))]
-    fn decrypt_record(&self, decryption_input: DecryptRecordInput) -> Result<String, RpcError>;
+    async fn decrypt_record(&self, decryption_input: DecryptRecordInput) -> Result<String, RpcError>;
 
     // todo: readd in Rust 1.54
     // #[cfg_attr(nightly, doc(include = "../documentation/private_endpoints/disconnect.md"))]
-    fn disconnect(&self, address: SocketAddr);
+    async fn disconnect(&self, address: SocketAddr) -> ();
```

Comment:
```
-> ()
```
this is by no means an issue, just wondering: did the compiler have some issue here without this explicit return value?